core/os/file/path.go

```diff
@@ -82,7 +83,7 @@ func (p Path) System() string { return p.value }
 
 // URL returns the full absolute path using the / separator prefixed by the URL
 // file scheme.
-func (p Path) URL() string { return "file:///" + filepath.ToSlash(p.value) }
+func (p Path) URL() (*url.URL, error) { return url.Parse("file:///" + filepath.ToSlash(p.value)) }
```

Comment:
In this particular case, if an error is raised, it is almost certainly an issue with the `file` package, not the user of the package. So, I'd say it is acceptable to drop the `error` return value and internally panic if `url.Parse` returns an error.